test/cruds/adding-row.spec.ts

```diff
@@ -337,3 +306,119 @@ describe('Adding row - column index', () => {
     expectArrayWithSameContent([2], index.getValueIndex(0, 0, 2).index)
   })
 })
+
+describe('Adding row - arrays', () => {
+  it('should be possible to add row above array', () => {
+    const engine = HyperFormula.buildFromArray([
+      ['=-C1:D3'],
+      [],
+      [],
+      ['foo']
+    ], {useArrayArithmetic: true})
+
+    engine.addRows(0, [0, 1])
+
+    const expected = HyperFormula.buildFromArray([
+      [],
+      ['=-C2:D4'],
+      [],
+      [],
+      ['foo']
+    ], {useArrayArithmetic: true})
+
+    expectEngineToBeTheSameAs(engine, expected)
+  })
+
+  it('adding row across array should not change array', () => {
+    const engine = HyperFormula.buildFromArray([
+      [], [], [],
+      ['=-A1:B3'],
+      [], [],
+      ['foo']
+    ], {useArrayArithmetic: true})
+
+    engine.addRows(0, [4, 1])
```

Comment:
Is it "across array"? Looks like after an array, and after the formula.

In excel when we add row within the range it will expand

Reply:
Maybe we miss one test, something like:
```suggestion
  it('adding row across array should not change array', () => {
    const engine = HyperFormula.buildFromArray([
      [], [], [],
      ['=-A1:B3'],
      [], [],
      ['foo']
    ], {useArrayArithmetic: true})

    engine.addRows(0, [2, 1])
```

Reply:
>Is it "across array"? 

It is. Above A1:B3 there is 3 rows.

> In excel when we add row within the range it will expand

But I do not add row within a range but across spilled array. And spilled array is never expanded that way.

Reply:
Ooohh, across spilled array, right! Now makes sense to me